src/mds/MDBalancer.cc

```diff
@@ -1058,6 +1128,74 @@ void MDBalancer::try_rebalance(balance_state_t& state)
   mds->mdcache->show_subtrees();
 }
 
+void MDBalancer::migrate_to_masked_rank()
```

Comment:
There is a decent amount of code duplication between this and try_rbalance() - I suggest we carve out the common parts if try_rebalance cannot be adjusted to integrate with migrating to a sub-set of active ranks.

Reply:
I will incorporate it into try_rebalance to eliminate duplicates.

Reply:
`migrate_to_masked_rank()` has been removed. The new approach is that if unmasked ranks are considered as exported, their target loads are set to zero. As a result, workloads might be migrated to maked ranks. https://github.com/ceph/ceph/pull/43284/files#diff-729d5135082091929c032d8d6a0552bd2a5c658006aa7cb409764d85e64f9431R890